cyctl/internal/create/template_auth_rules.go

```diff
@@ -0,0 +1,112 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cyclops-ui/cyclops/cyclops-ctrl/api/v1alpha1"
+	"github.com/cyclops-ui/cyclops/cyclops-ctrl/api/v1alpha1/client"
+	"github.com/cyclops-ui/cycops-cyctl/internal/kubeconfig"
+	"github.com/spf13/cobra"
+	v1Spec "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	createTemplateAuthRuleExample = ` 
+	# Create templateauth rule
+	cyctl create templateauthrule NAME --repo='github.com/repo/a' --username='name:key' --password='name:key'
+ 
+	# Create templateauthrule
+	cyctl create templateauthrule demo-templateauthrule --repo='https://github.com/cyclops-ui/templates' --username='name:john' --password='name:random'`
+)
+
+var (
+	username string
+	password string
+)
+
+// createTemplateAuthRule allows you to create create TemplateAuthRule Custom Resource.
+func createTemplateAuthRule(clientset *client.CyclopsV1Alpha1Client, templateAuthRuleName string) {
+	_ = clientset
+	usernameSlice := strings.Split(username, ":")
+	passwordSlice := strings.Split(password, ":")
+
+	// Ensure key is present for both username and password
+	if len(usernameSlice) < 2 || usernameSlice[1] == "" || len(passwordSlice) < 2 || passwordSlice[1] == "" {
+		fmt.Printf("Invalid format for username or password. Expected 'name:key'.\n")
+		return
+	}
```

Comment:
Is it valid to set `:key`? I'd say no since you don't know from which secret is it coming from

Reply:
Yeah sure, i'll checkout by end of this week